databuilder/databuilder/extractor/neo4j_extractor.py

```diff
@@ -66,15 +67,29 @@ def _get_driver(self) -> Any:
         """
         Create a Neo4j connection to Database
         """
-        trust = neo4j.TRUST_SYSTEM_CA_SIGNED_CERTIFICATES if self.conf.get_bool(Neo4jExtractor.NEO4J_VALIDATE_SSL) \
-            else neo4j.TRUST_ALL_CERTIFICATES
-        return GraphDatabase.driver(uri=self.graph_url,
-                                    max_connection_lifetime=self.conf.get_int(
-                                        Neo4jExtractor.NEO4J_MAX_CONN_LIFE_TIME_SEC),
-                                    auth=(self.conf.get_string(Neo4jExtractor.NEO4J_AUTH_USER),
-                                          self.conf.get_string(Neo4jExtractor.NEO4J_AUTH_PW)),
-                                    encrypted=self.conf.get_bool(Neo4jExtractor.NEO4J_ENCRYPTED),
-                                    trust=trust)
+        # The config settings 'encrypted' and 'trust' can only be used with the URI
+        # schemes ['bolt', 'neo4j'].
+        uri = self.graph_url
+        if uri.startswith('bolt:') or uri.startswith('neo4j:'):
+            trust = neo4j.TRUST_SYSTEM_CA_SIGNED_CERTIFICATES if self.conf.get_bool(Neo4jExtractor.NEO4J_VALIDATE_SSL) \
+                else neo4j.TRUST_ALL_CERTIFICATES
+            driver = \
+                GraphDatabase.driver(uri=uri,
```

Comment:
not clear why this is necessary

Reply:
oh I see https://github.com/neo4j/neo4j-python-driver#connecting-with-python-driver-4x-to-neo4j-35

Reply:
i meant in general having the two-sections does not make sense to me yet. we can chat offline.